Monika After Story/game/zz_hangman.rpy

```diff
@@ -496,18 +489,62 @@ label mas_hangman_game_loop:
         m "[hm_hint]"
 
     # main loop for hangman game
-    $ done = False
-    $ win = False
-    $ chances = 6
-    $ guesses = 0
-    $ missed = ""
-    $ avail_letters = list(hm_ltrs_only)
-    $ give_up = False
-
-    if persistent._mas_sensitive_mode:
-        $ avail_letters.remove("?")
-
-    $ dt_color = mas_hmg.WORD_COLOR
+    python:
+        done = False
+        win = False
+        chances = 6
+        guesses = 0
+        last_chance_guesses = 0
+        missed = ""
+        avail_letters = list(hm_ltrs_only)
+        give_up = False
+
+        if persistent._mas_sensitive_mode:
+            avail_letters.remove("?")
+
+        dt_color = mas_hmg.WORD_COLOR
+
+        def getCorrectGuesses(guesses, chances, max_chances=6):
+            """
+            Get the amount of correct guesses, i.e. the amount of guesses 
+            that actually exposed letters in the word.
+
+            IN:
+                guesses - amount of guesses (including incorrect) made
+                    by the player
+                chances - amount of chances the player has
+                max_chances - maximum amount of chances player can
+                    have (defaults to 6)
+
+            RETURNS:
+                Amount of correct guesses.
+            """
+            return guesses + chances - max_chances
+
+        def getLastChanceGuessesRatio(guesses, chances, last_chance_guesses,
+                                      useCorrect=False):
```

Comment:
please don't use this format. Keep this all in a single line or all on their own line

Reply:
What exactly do you mean by "define `full_wordlist` on MAS level"? I don't feel like I understood it the right way.

Reply:
No problem.